src/v2/providers/https.ts

```diff
@@ -265,12 +265,19 @@ export function onRequest(
       // Respect `cors: false` to turn off cors even if debug feature is enabled.
       origin = opts.cors === false ? false : true;
     }
+    // Arrays cause the access-control-allow-origin header to be dynamic based
+    // on the origin header of the request. If there is only one element in the
+    // array, this is unnecessary.
+    if (Array.isArray(origin) && origin.length === 1) {
+      origin = origin[1];
```

Comment:
Is this a bug... shouldn't this either be `origin = origin[0]` or `origin = undefined`?

Reply:
🤬 Will fix this week